salt/transport/zeromq.py

```diff
@@ -736,6 +736,17 @@ def publish(self, load):
         if load['tgt_type'] == 'list':
             int_payload['topic_lst'] = load['tgt']
 
+        # If zmq_filtering is enabled, target matching has to happen master side
+        if self.opts['zmq_filtering']:
```

Comment:
Our specific use case is PCRE match of minion-id. As a step in the right direction, what about only supporting the things you mentioned master side knows about: minion-id, and pillar? (I have not used anything beyond minion-id to be honest) And dropping the other match types on the broadcast channel.